src/bot.ts

```diff
@@ -142,17 +142,22 @@ export class DiscordBot {
         if (this.channelLocks[channel.id]) {
             return;
         }
-        let i: NodeJS.Timeout;
+
         const p = new Promise((resolve) => {
-            i = setInterval(resolve, this.config.limits.discordSendDelay);
-            this.channelLocks[channel.id] = {i, p};
+            if (!this.channelLocks[channel.id]) {
+                return;
```

Comment:
resolve(); before return, just in case?

Reply:
The return is designed to save us setting up a timer if we have already unlocked the channel before the promise is processed. It's astronomically unlikely, but we don't need to setup a timer if it's been deleted.

Reply:
soru meany
```js
if (!this.channelLocks[channel.id]) {
    resolve();
    return;
}
```
that the promise still resolves if we don't need a timer

Reply:
ooooh, yes